collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2632,6 +2629,15 @@ public void onCancelFormLoading() {
         finish();
     }
 
+    private void onDataChanged(Object data) {
+        ODKView odkView = getCurrentViewIfODKView();
+        if (odkView != null) {
+            QuestionWidget widgetGettingNewValue = getWidgetWaitingForBinaryData();
+            setBinaryWidgetData(data);
```

Comment:
Please rename this method to `setWidgetData()` since like in this case it's not only about binary files.